Extensions/GDevelopAuthentication/gdevelopauthenticationtools.ts

```diff
@@ -0,0 +1,865 @@
+namespace gdjs {
+  declare var cordova: any;
+
+  const logger = new gdjs.Logger('Authentication');
+  export namespace gdevelopAuthentication {
+    // Authentication information.
+    let _username: string | null = null;
+    let _userId: string | null = null;
+    let _userToken: string | null = null;
+    let _justLoggedIn = false;
+
+    let _error: any = null;
+    let _checkedLocalStorage: boolean = false;
+
+    // Authentication display
+    let _authenticationIframe: HTMLIFrameElement | null = null; // For Electron.
+    let _authenticationWindow: Window | null = null; // For Web.
+    let _authenticationInAppWindow: Window | null = null; // For Cordova.
+    let _authenticationRootContainer: HTMLDivElement | null = null;
+    let _authenticationSubContainer: HTMLDivElement | null = null;
+    let _authenticationLoader: HTMLDivElement | null = null;
+    let _authenticationBanner: HTMLDivElement | null = null;
+    let _authenticationTimeoutId: NodeJS.Timeout | null = null;
+    let _authenticationMessageCallback:
+      | ((event: MessageEvent) => void)
+      | null = null;
+    let _cordovaAuthenticationMessageCallback:
+      | ((event: MessageEvent) => void)
+      | null = null;
+
+    /**
+     * Returns true if a user token is present in the local storage.
+     */
+    export const isAuthenticated = () => {
+      if (!_checkedLocalStorage) {
+        fetchAuthenticatedUserFromLocalStorage();
+      }
+      return _userToken !== null;
+    };
+
+    /**
+     * Returns true if the user just logged in.
+     * Useful to update username or trigger messages in the game.
+     */
+    export const hasLoggedIn = () => {
+      if (_justLoggedIn) {
+        _justLoggedIn = false;
+        return true;
+      }
+      return false;
+    };
+
+    /**
+     * Returns the username from the local storage.
+     */
+    export const getUsername = () => {
+      if (!_checkedLocalStorage) {
+        fetchAuthenticatedUserFromLocalStorage();
+      }
+      return _username || '';
+    };
+
+    /**
+     * Remove the user information from the local storage.
+     */
+    export const logout = (runtimeScene: RuntimeScene) => {
+      if (!_checkedLocalStorage) {
+        fetchAuthenticatedUserFromLocalStorage();
+      }
+      _username = null;
+      _userToken = null;
+
+      const _gameId = gdjs.projectData.properties.projectUuid;
+      (window as any).localStorage.removeItem(`${_gameId}_authenticatedUser`);
+      _authenticationBanner = computeNotAuthenticatedBanner(runtimeScene);
+    };
+
+    /**
+     * Retrieves the user information from the local storage, and store
+     * them in the extension variables.
+     * Returns if the fetched user is different than the current one used
+     * in the variables. (useful when the user changes)
+     */
+    const fetchAuthenticatedUserFromLocalStorage = (): boolean => {
```

Comment:
Should it be renamed `readAuthenticatedUserInLocalStorage`: I was expecting an API call to be made